File: Python/Factorial.py

def fact(n):
   if n==0:
      return 1
   return n*fact(n-1)



# To take input from the user
n = int(input("Enter a number: "))

factorial = 1

# checking wheather the number is negative, positive or zero
if n < 0:
   print("factorial does not exist for negative numbers")
elif n == 0:
   print("The factorial of 0 is 1")
else:
#    for i in range(1,num + 1):
#        factorial = factorial*i
#    print("The factorial of",n,"is",factorial)

     print("factorial of the number is :"+fact(n))

File: Data Structures and Algorithms/Algorithms/c/Factorial.c
#include<stdio.h>

int factorial(int number)
{
    if(number == 0)
        return 1;
    else
        return(number*factorial(number - 1));
}
int main()
{
    int n, num;
    printf("Enter no. to find factorial: ");
    scanf("%d", &n);
    num = factorial(n);
    printf("%d! = %d", n, num);
}
